### ISSUE TITLE
Condor jobs not auto-removed

### ISSUE THREAD

**Issue by yadudoc** [Parsl/parsl#1427]
**Describe the bug**
`CondorProvider` sets a `leave_in_queue=True` flag that causes condor jobs to linger in the queue
indefinitely. This might cause issues in cases where total jobs in queue are limited.

**To Reproduce**
Steps to reproduce the behavior, for e.g:
1. Setup Parsl 0.8.0 with Python 3.6 on cluster
2. Run tests with CondorProvider
3. Wait to complete
4. condor_q will show tasks in 

**Expected behavior**
Condor jobs should be removed once Parsl finishes execution.

**Environment**
?

Reported by Matthew Becker from the DESC community.

**Comment by hategan** [comment]
Not sure. In many cases the queuing system attaches errors to the job status that are otherwise not accessible. This is an incentive to actually poll the queue for errored jobs when possible. So we might not want to completely abandon that.

I think Swift did something similar by setting leave_in_queue=True until a terminal status was detected, after which it would set leave_in_queue to False. But it seems like that scheme would also leave jobs in the queue when things were abruptly terminated.

**Comment by benclifford** [comment]
leave_in_queue is can be any ClassAd expression that evaluates to a bool, not just a literal True/False.

Maybe there's something interesting that can be done there to attempt to preserve jobs for a while but make them disappear eventually - eg keep for n mins after completion according to ClassAds, but still deleted by parsl when it no longer needs the jobs to be there?


As an example of a more complicated expression, https://research.cs.wisc.edu/htcondor/manual/v8.7.8/labelmancondorsubmitCondorsubmit.html gives:

```
leave_in_queue = (JobStatus == 4) && ((StageOutFinish =?= UNDEFINED) ||\  
                 (StageOutFinish == 0))
```

**Comment by benclifford** [comment]
Eg leave_in_queue could go false when completion time is more than 3x the parsl polling period.

**Comment by hategan** [comment]
Yeah, that's what I thought many years ago, but couldn't figure out a solution, but maybe I didn't really try. I see:
on_exit_hold = (time() - JobStartDate) < (60 * $(MINUTE))

So maybe leave_in_queue = (time() - JobEndDate) < (60 * $(MINUTE))
